src/main/java/net/snowflake/client/core/SFSession.java

```diff
@@ -656,6 +656,11 @@ public synchronized void open() throws SFException, SnowflakeSQLException {
     } else {
       TelemetryService.disableHTAP();
     }
+    if (isClientTelemetryEnabled()) {
```

Comment:
inbound telemetry is enabled per session - but here we set it on the static level - do we need it at all? can we just use the flag from the session wherever it's necessary?